tests/system/aiplatform/test_e2e_vertex_model.py

```diff
@@ -0,0 +1,149 @@
+            self.train_loop(dataloader, loss_fn, optimizer)
+
+    def predict(self, data):
+        return self.forward(data)
+
+    # Implementation of predict_payload_to_predict_input(), which converts a predict_payload object to predict() inputs
+    def predict_payload_to_predict_input(self, instances):
+        feature_columns = ["feat_1", "feat_2"]
+        data = pd.DataFrame(instances, columns=feature_columns)
+        torch_tensor = torch.tensor(data[feature_columns].values).type(
+            torch.FloatTensor
+        )
+        return torch_tensor
+
+    # Implementation of predict_input_to_predict_payload(), which converts predict() inputs to a predict_payload object
+    def predict_input_to_predict_payload(self, parameter):
+        return parameter.tolist()
+
+    # Implementation of predict_output_to_predict_payload(), which converts the predict() output to a predict_payload object
+    def predict_output_to_predict_payload(self, output):
+        return output.tolist()
+
+    # Implementation of predict_payload_to_predict_output, which takes a predict_payload object containing predictions and
+    # converts it to the type of output expected by the user-written class.
+    def predict_payload_to_predict_output(self, predictions):
+        data = pd.DataFrame(predictions)
+        torch_tensor = torch.tensor(data.values).type(torch.FloatTensor)
+        return torch_tensor
+
+
+@pytest.mark.usefixtures("prepare_staging_bucket", "delete_staging_bucket", "teardown")
+class TestEndToEndVertexModel(e2e_base.TestEndToEnd):
+    """End to end system test of the Vertex SDK with the VertexModel interface"""
+
+    _temp_prefix = "temp-vertex-sdk-e2e-vertex-model"
+
+    def test_end_to_end_vertex_model(self, shared_state):
+        # Collection of resources generated by this test, to be deleted during teardown
+        shared_state["resources"] = []
+
+        aiplatform.init(
+            project=e2e_base._PROJECT,
+            location=e2e_base._LOCATION,
+            staging_bucket=shared_state["staging_bucket_name"],
+        )
+
+        # Set up random training data
+
+        df = pd.DataFrame(
+            np.random.random(size=(100, 3)), columns=["feat_1", "feat_2", "target"]
+        )
+
+        shared_state["resources"].extend([df])
+
+        # Train using VertexModel interface
+
+        my_model = LinearRegression(2, 1)
+        my_model.remote = True
+
+        my_model.fit(df, "target", 1, 0.1)
+
+        shared_state["resources"].extend([my_model._model])
+
+        # Set up test data
+
+        data = pd.DataFrame(
+            np.random.random(size=(100, 3)), columns=["feat_1", "feat_2", "target"]
+        )
+        feature_columns = list(data.columns)
+        feature_columns.remove("target")
+        torch_tensor = torch.tensor(data[feature_columns].values).type(
+            torch.FloatTensor
+        )
+
+        # Prediction with remotely-trained model
+
+        predictions = my_model.predict(torch_tensor)
```

Comment:
The created Endpoint should be added to `shared_state['resources']` as well.

Reply:
got it!